pkg/reconciler/managed/reconciler.go

```diff
@@ -189,37 +188,32 @@ type ExternalConnecter interface {
 	Connect(ctx context.Context, mg resource.Managed) (ExternalClient, error)
 }
 
-// An ExternalDisconnecter disconnect from provider.
+// An ExternalDisconnecter disconnects from a provider.
 type ExternalDisconnecter interface {
-	// Disconnect from the provider and close an ExternalClient.
+	// Disconnect from the provider and close the ExternalClient.
 	Disconnect(ctx context.Context) error
 }
 
-// NopDisconnect does nothing. Never returns error.
-func NopDisconnect(_ context.Context) error {
-	return nil
-}
-
-// externalConnectDisconnecter is wrap ExternalConnecter in externalConnectDisconnecter
-// with nop Disconnect.
-type externalConnectDisconnecter struct {
+// A NopDisconnecter converts an ExternalConnecter into an
+// ExternalConnectDisconnecter with a no-op Disconnect method.
+type NopDisconnecter struct {
```

Comment:
FWIW I took a pass at a commit that fixes the various 'connecter' spelling mistakes, changing them to 'connector' mostly using type aliases. I decided to abandon it since the mistake has been there forever, and marking the incorrect spelling deprecated would cause quite a lot of downstream churn.

Reply:
I actually didn't even realize that was a mistake -- I believe they are both words right?

Reply:
https://www.askdifference.com/connector-vs-connecter/ - Huh I guess so. In that case we're inconsistent but at least nothing is incorrect haha.